elementary/clients/dbt/slim_dbt_runner.py

```diff
@@ -23,15 +27,42 @@
 # Disable dbt tracking
 disable_tracking()
 
-DEFAULT_VARS = "{}"
+dbt_version = version.parse(dbt_version_string)
+
+DEFAULT_VARS: Union[str, Dict[str, Any]]
+if dbt_version >= version.parse("1.5.0"):
+    DEFAULT_VARS = {}
+else:
+    DEFAULT_VARS = "{}"
+
+
+def default_project_dir() -> Path:
+    if "DBT_PROJECT_DIR" in os.environ:
+        return Path(os.environ["DBT_PROJECT_DIR"]).resolve()
+    paths = list(Path.cwd().parents)
+    paths.insert(0, Path.cwd())
+    return next((x for x in paths if (x / "dbt_project.yml").exists()), Path.cwd())
+
+
+def default_profiles_dir() -> Path:
+    if "DBT_PROFILES_DIR" in os.environ:
+        return Path(os.environ["DBT_PROFILES_DIR"]).resolve()
+    return (
+        Path.cwd() if (Path.cwd() / "profiles.yml").exists() else Path.home() / ".dbt"
+    )
+
+
+DEFAULT_PROFILES_DIR = str(default_profiles_dir())
+DEFAULT_PROJECT_DIR = str(default_project_dir())
 
 
 class ConfigArgs(BaseModel):
-    profiles_dir: Optional[str] = None
-    project_dir: str
+    project_dir: str = DEFAULT_PROJECT_DIR
+    profiles_dir: str = DEFAULT_PROFILES_DIR
+    profile: Optional[str] = None
     target: Optional[str] = None
     threads: Optional[int] = 1
-    vars: Optional[str] = DEFAULT_VARS
+    vars: Optional[Union[str, Dict[str, Any]]] = DEFAULT_VARS
```

Comment:
Wouldn't it make more sense to always have that as a `dict` and do `json.dumps()` based on the version?

Reply:
This structure is passed to dbt, and vars need to be either a dict or a string depending on the version, so I must declare it as a union